articles/api-auth/dynamic-client-registration.md

```diff
@@ -0,0 +1,205 @@
+
+### Promote connections
+
+Clients registered via the [Dynamic Client Registration Endpoint](#dynamic-client-registration-endpoint) are flagged as **Third Party Clients** and can only authenticate users using connections flagged as **Domain Connections**. These connections will be open for any dynamic client to allow users to authenticate.
+
+You can promote a connection to domain level using the [Update a Connection endpoint](/api/management/v2#!/Connections/patch_connections_by_id).
+
+```
+curl -X PATCH -H "Content-Type: application/json" -H "Authorization: Bearer API2_ACCESS_TOKEN" -H "Cache-Control: no-cache" -d '{
+  "is_domain_connection": true
+}' "https://${account.namespace}/api/v2/connections/CONNECTION_ID"
+```
+
+Where:
+- `API2_ACCESS_TOKEN`: [Α valid Auth0 API2 token](/api/management/v2/tokens#how-to-get-a-management-apiv2-token) with the scope `update:connections`.
+- `CONNECTION_ID`: Τhe Id of the connection to be promoted.
+
+
+### Update the login page
+
+If you use or would like to use an Auth0 [Hosted Login Page](/hosted-pages/login) with the Dynamic Client feature, you need to use at least version `10.7.x` of Lock, and set `__useTenantInfo: config.isThirdPartyClient` when instantiating Lock.
+
+Sample script:
+
+```html
+<script src="https://cdn.auth0.com/js/lock/10.7/lock.min.js"></script>
+...
+<script>
+  // Decode utf8 characters properly
+  var config = JSON.parse(decodeURIComponent(escape(window.atob('@@config@@'))));
+
+  var connection = config.connection;
+  var prompt = config.prompt;
+  var languageDictionary;
+  var language;
+  if (config.dict && config.dict.signin && config.dict.signin.title) {
+    languageDictionary = { title: config.dict.signin.title };
+  } else if (typeof config.dict === 'string') {
+    language = config.dict;
+  }
+
+  var lock = new Auth0Lock(config.clientID, config.auth0Domain, {
+    auth: {
+      redirectUrl: config.callbackURL,
+      responseType: config.callbackOnLocationHash ? 'token' : 'code',
+      params: config.internalOptions
+    },
+    assetsUrl:  config.assetsUrl,
+    allowedConnections: connection ? [connection] : null,
+    rememberLastLogin: !prompt,
+    language: language,
+    languageDictionary: languageDictionary,
+    closable: false,
+    __useTenantInfo: config.isThirdPartyClient
+  });
+
+  lock.show();
+</script>
+```
+
+## Use dynamic registration
+
+In this section we will see how a third party developer can register and configure a client.
+
+### Register your client
+
+In order to dynamically register a client with Auth0, you need to send an HTTP `POST` message to the [Client Registration endpoint](/api/management/v2#!/oidc/register). Note that Auth0 supports open Dynamic Registration, which means that the [Client Registration endpoint](/api/management/v2#!/oidc/register) will accept a registration request without an [OAuth 2.0 Access Tokens](/tokens/access-token).
```

Comment:
This link does not exist `/api/management/v2#!/oidc/register`. We don't have swagger for this endpoint.